ntcore/src/main/native/cpp/local/LocalStorageImpl.cpp

```diff
@@ -0,0 +1,1086 @@
+        topic->name);
+    return nullptr;
+  }
+
+  // Create subscriber
+  auto subscriber =
+      AddLocalSubscriber(topic, PubSubConfig{type, typeStr, options});
+
+  // Create entry
+  return AddEntry(subscriber);
+}
+
+LocalEntry* StorageImpl::GetEntry(std::string_view name) {
+  if (name.empty()) {
+    return nullptr;
+  }
+
+  // Get the topic data
+  auto* topic = GetOrCreateTopic(name);
+
+  if (!topic->entry) {
+    if (topic->localSubscribers.size() >= kMaxSubscribers) {
+      ERR("reached maximum number of subscribers to '{}', not creating entry",
+          topic->name);
+      return nullptr;
+    }
+
+    // Create subscriber
+    auto* subscriber = AddLocalSubscriber(topic, {});
+
+    // Create entry
+    topic->entry = AddEntry(subscriber);
+  }
+
+  return topic->entry;
+}
+
+void StorageImpl::RemoveSubEntry(NT_Handle subentryHandle) {
+  Handle h{subentryHandle};
+  if (h.IsType(Handle::kSubscriber)) {
+    RemoveLocalSubscriber(subentryHandle);
+  } else if (h.IsType(Handle::kMultiSubscriber)) {
+    RemoveMultiSubscriber(subentryHandle);
+  } else if (h.IsType(Handle::kEntry)) {
+    if (auto entry = RemoveEntry(subentryHandle)) {
+      RemoveLocalSubscriber(entry->subscriber->handle);
+      if (entry->publisher) {
+        RemoveLocalPublisher(entry->publisher->handle);
+      }
+    }
+  }
+}
+
+std::unique_ptr<LocalPublisher> StorageImpl::RemoveLocalPublisher(
+    NT_Publisher pubHandle) {
+  auto publisher = m_publishers.Remove(pubHandle);
+  if (publisher) {
+    auto topic = publisher->topic;
+    bool didExist = topic->Exists();
+    topic->localPublishers.Remove(publisher.get());
+    if (didExist && !topic->Exists()) {
+      topic->ResetIfDoesNotExist();
+      NotifyTopic(topic, NT_EVENT_UNPUBLISH);
+    }
+
+    if (publisher->active && m_network) {
+      m_network->ClientUnpublish(Handle{publisher->handle}.GetIndex());
+    }
+
+    if (publisher->active && !topic->localPublishers.empty()) {
+      // some other publisher still exists; if it has a different type, refresh
+      // and publish it over the network
+      auto& nextPub = topic->localPublishers.front();
+      if (nextPub->config.type != topic->type ||
+          nextPub->config.typeStr != topic->typeStr) {
+        topic->type = nextPub->config.type;
+        topic->typeStr = nextPub->config.typeStr;
+        RefreshPubSubActive(topic, false);
+        if (nextPub->active && m_network) {
+          m_network->ClientPublish(Handle{nextPub->handle}.GetIndex(),
+                                   topic->name, topic->typeStr,
+                                   topic->properties, nextPub->config);
+        }
+      }
+    }
+  }
+  return publisher;
+}
+
+//
+// Multi-subscriber functions
+//
+
+LocalMultiSubscriber* StorageImpl::AddMultiSubscriber(
+    std::span<const std::string_view> prefixes, const PubSubOptions& options) {
+  DEBUG4("AddMultiSubscriber({})", fmt::join(prefixes, ","));
```

Comment:
This debug will be emitted even if we've reached the maximum number of multisubscribers. (This affects `LocalStorage::SubscribeMultiple` and `LocalStorage::AddListener(NT_Listener, std::span<const std::string_view>, unsigned int)`) Not necessarily bad, but technically a change in behavior.

Reply:
It's a debug print, so that's fine.